probe-dictionary/explore.js

```diff
@@ -593,10 +603,42 @@ function updateSearchParams(pushState = false) {
     params.detailView = gDetailViewId;
   }
 
+  if (params['search']=="") {
+    delete params.search;
+  }
+  if (params['searchtype']=="in_any") {
+    delete params.searchtype;
+  }
+  if (params['optout']==false) {
+    delete params.optout;
+  }
+  if (params['channel']=="any") {
+    delete params.channel;
+  }
+  if (params['constraint']=="is_in") {
+    delete params.constraint;
+  }
+  if (params['version']=="any") {
+    delete params.version;
+  }
+  if (params['view']=="search-results-view") {
+    delete params.view;
+  }
+  
   if (!pushState) {
-    window.history.replaceState("", "", "?" + $.param(params));
```

Comment:
This is a bit of an ornate way of doing things. I think we can simplify this with something like
```js
let queryString = "";
if (Object.keys(params).length) {
  queryString = "?" + $.param(params);
}
```

What do you think?

Reply:
@chutten , Yeah, sounds very right ! I will change it now. Thanks!